ftplugin/markdown.vim

```diff
@@ -23,7 +23,7 @@ function! MarkdownFold()
   let line = getline(v:lnum)
 
   " Regular headers
-  if (match(line, '^#\+ .*$') == 0)
+  if (line =~ '^#\+ .*$')
```

Comment:
`=~#`, so the linter doesn't get mad (and you can drop the outer parentheses).

Reply:
And `.*$` is redundant (but the old code had the same redundancy, so that one's on me).

Reply:
Sorry for missing the `#`, fixed.

I've also removed the redundant `.*$`.